pkg/apis/mattermost/v1alpha1/clusterinstallation_utils.go

```diff
@@ -120,16 +120,45 @@ func (mattermost *ClusterInstallation) GenerateIngress() *v1beta1.Ingress {
 }
 
 // GenerateDeployment returns the deployment spec for Mattermost
-func (mattermost *ClusterInstallation) GenerateDeployment(dbUser, dbPassword string) *appsv1.Deployment {
-	datasourceMM := fmt.Sprintf("mysql://root:%s@tcp(%s-mysql.%s:3306)/mattermost?charset=utf8mb4,utf8&readTimeout=30s&writeTimeout=30s", dbPassword, mattermost.Name, mattermost.Namespace)
+func (mattermost *ClusterInstallation) GenerateDeployment(dbUser, dbPassword string, externalDB bool) *appsv1.Deployment {
+	initCheckDB := corev1.Container{}
+	initDB := corev1.Container{}
+	envVarDB := corev1.EnvVar{
+		Name: "MM_CONFIG",
+	}
 
-	initContainerCMD := fmt.Sprintf("until curl --max-time 5 http://%s-mysql.%s:3306; do echo waiting for mysql; sleep 5; done;", mattermost.Name, mattermost.Namespace)
-	cmdInit := []string{"sh", "-c"}
-	cmdInit = append(cmdInit, initContainerCMD)
-	cmdStartMM := []string{"mattermost"}
+	if externalDB {
+		secretName := fmt.Sprintf("%s-externalDB", mattermost.Name)
+		envVarDB.ValueFrom = &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: secretName,
+				},
+				Key: "externalDB",
+			},
+		}
+	} else {
+		datasourceMM := fmt.Sprintf("mysql://%s:%s@tcp(%s-mysql.%s:3306)/mattermost?charset=utf8mb4,utf8&readTimeout=30s&writeTimeout=30s", dbUser, dbPassword, mattermost.Name, mattermost.Namespace)
+		envVarDB.Value = datasourceMM
+
+		// Create the init container to check with the DB is up and running
+		initContainerCMD := fmt.Sprintf("until curl --max-time 5 http://%s-mysql.%s:3306; do echo waiting for mysql; sleep 5; done;", mattermost.Name, mattermost.Namespace)
```

Comment:
What happens if mysql fails to come up? Is the deployment "stuck", or can it be recovered by trying again?

Reply:
if I'm not wrong the pod will be restarted because the readiness probe did not succeed